engine/src/main/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionProcessor.java

```diff
@@ -53,21 +53,23 @@ public ResourceDeletionProcessor(
   @Override
   public void processNewCommand(final TypedRecord<ResourceDeletionRecord> command) {
     final var value = command.getValue();
-    tryDeleteResources(command);
-
     final long eventKey = keyGenerator.nextKey();
     stateWriter.appendFollowUpEvent(eventKey, ResourceDeletionIntent.DELETING, value);
-    responseWriter.writeEventOnCommand(eventKey, ResourceDeletionIntent.DELETING, value, command);
-    stateWriter.appendFollowUpEvent(eventKey, ResourceDeletionIntent.DELETED, value);
 
+    tryDeleteResources(command);
+
+    stateWriter.appendFollowUpEvent(eventKey, ResourceDeletionIntent.DELETED, value);
     commandDistributionBehavior.distributeCommand(eventKey, command);
+    responseWriter.writeEventOnCommand(eventKey, ResourceDeletionIntent.DELETING, value, command);
```

Comment:
👍 I like having the response at the end, this could be a good practice for all our processor 🤔